nibabel/gifti/gifti.py

```diff
@@ -680,6 +680,45 @@ def get_arrays_from_intent(self, intent):
         it = intent_codes.code[intent]
         return [x for x in self.darrays if x.intent == it]
 
+    def agg_data(self, intent_code=None):
+        """
+        Retrun a numpy arrary of aggregated GiftiDataArray of the same intent code
+        or
+        Retrun GiftiDataArray in tuples for surface files
+
```

Comment:
It would be good to add some doctests that demonstrate how to use this. If we include a time series and a surface file, we could follow the examples of `get_fdata()`:

https://github.com/nipy/nibabel/blob/22fe8c20de0c92456a279627a70b22c3fb84dd16/nibabel/dataobj_images.py#L287-L342

I would show aggregating data:

* without intent code
* with matching intent codes
* with mismatching intent codes (should return `()`)
* with tuple intent codes